Automattic/jetpack
# tiled-gallery: Do not photonize images for private+atomic blogs

Improves https://github.com/Automattic/wp-calypso/issues/41272# 

#### Changes proposed in this Pull Request:
* **The tiled-gallery block will not "photonize" images when used on sites that are both private and atomic.**
  * The photon CDN cannot handle authentication. Previously, it was never able to load these images, meaning users would see blank rectangles after adding images to a tiled-gallery on a private atomic site.
  * This is a stop-gap that allows the images to show, but does not support the resizing that the tiled-gallery block uses to save bandwidth. However, showing something is better than showing nothing.
  * This may later be improved upon further by "Proxied Images", or teaching the `<GalleryImageEdit>` and `<GalleryImageSave>` components inside tiled-gallery block to make XHR requests to public-api, download private+atomic image blobs, then render those to the user. See discussion in https://github.com/Automattic/wp-calypso/issues/41272.
    * This trick is from Calypso, which uses it here: [Link](https://github.com/Automattic/wp-calypso/blob/ef76ff4b6871663ced67b361ae17fac301f5248e/client/my-sites/media-library/proxied-image.tsx#L77-L81)
    * Could there be another way to solve this problem?

#### Jetpack product discussion
* None so far.
 
### Does this pull request change what data or activity we track or use?
* No

#### Testing instructions (Initial Setup):






* Check out this branch.
* Set up a jetpack development environment with docker and JT (jurassic-tube, ngrok-like service for A8Cs) ( PCYsg-efV-p2 )
* Edit the docker volume configuration to allow for image uploads. (Needed on linux only? Worth checking if image uploads work w/o this step)
  * Create the `tools/docker/uploads` directory and chmod 777 it: `cd tools/docker ; mkdir uploads ; chmod 777 uploads`.
  * Add this line to `tools/docker/compose-volumes.yml`: 
```
- ./uploads:/var/www/html/wp-content/uploads
```
* (Optional) Reset the docker blog. `yarn docker:clean`, `yarn docker:up`, `yarn docker:install`. Note that the docker:install command assumes you have a `tools/docker/default.env` file with the l/p you want. Or you can install it manually if you would like by visiting http://localhost and filling the prompts.
* Run the docker environment. In three tabs:
  * `yarn docker:up`
  * `yarn docker:jt-up`
  * `cd projects/plugins/jetpack ; yarn build-extensions --watch`
* Ensure the blog is connected to WordPress.com. `/wp-admin/admin.php?page=jetpack#/my-plan` should not show "Offline Mode". If it does, visit `/wp-admin/` and click the green connect Jetpack button, click approve, click start for free.

#### Testing instructions (Make your jetpack site think it's atomic)
* Manually edit [this line](https://github.com/Automattic/jetpack/blob/b967d425861e8da56a1b37edc786c9fa475d5fed/projects/plugins/jetpack/extensions/shared/site-type-utils.js#L37) to be a true value: 
```
export function isAtomicSite() {
	return true;
}
```

#### Testing instructions (Verify Images Continue to be Photonized for Public Blogs):
* Ensure the blog is public
  * Visit http://localhost/wp-admin/options.php . `blog_public` should be `0` or `1`.
* Go to media library and upload 2 images.
* Create a new post
* Add a tiled-gallery block
* Add the two images
* Inspect element on the two images and look at the `srcset`. They should have urls like `i1.wp.com` (number may vary). This means they are still being photonized, which is the default behavior.
* Public the post and view the post outside the editor.
* Inspect element on the two images and look at the `srcset`. They should have urls like `i1.wp.com` (number may vary). This means they are still being photonized, which is the default behavior.

#### Testing instructions (Verify Images are not Photonized for Private Blogs):
* Ensure the blog is private
  * Visit http://localhost/wp-admin/options.php . Change `blog_public` to `-1`.
* Q: "Hey, but the other condition is Atomic..."
  * A: Yes. Terribly sorry, but I don't know how to test this. The PR is written with  `'isAtomicSite'  => true` inside `tiled-gallery.php`, to hardcode that condition, which should be replaced with `'isAtomicSite'  => jetpack_is_atomic_site(),` before merging.
* Create a new post
* Add a tiled-gallery block
* Add the two images
* Inspect element on the two images and look at the `srcset`. They should **NOT** have urls like `i1.wp.com` (number may vary). This means they are not being photonized.
* Public the post and view the post outside the editor.
* Inspect element on the two images and look at the `srcset`. They should **NOT** have urls like `i1.wp.com` (number may vary). This means they are not being photonized.
